src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -64,6 +60,28 @@ void GcsServer::Start() {
       main_service_, redis_client_->GetPrimaryContext(), [this]() { Stop(); });
   gcs_redis_failure_detector_->Start();
 
+  // Init gcs table storage.
+  if (RayConfig::instance().gcs_storage() == "redis") {
+    gcs_table_storage_ = std::make_shared<gcs::RedisGcsTableStorage>(redis_client_);
+  } else if (RayConfig::instance().gcs_storage() == "memory") {
+    gcs_table_storage_ = std::make_shared<InMemoryGcsTableStorage>(main_service_);
+  } else {
+    RAY_LOG(FATAL) << "Unsupported gcs storage: " << RayConfig::instance().gcs_storage();
+  }
+  // Init internal config
+  auto on_done = [this](const ray::Status &status) {
+    RAY_CHECK(status.ok()) << "Failed to put internal config";
+    this->main_service_.stop();
+  };
+  ray::rpc::StoredConfig stored_config;
+  stored_config.set_config(config_.config_list);
+  RAY_CHECK_OK(gcs_table_storage_->InternalConfigTable().Put(ray::UniqueID::Nil(),
+                                                             stored_config, on_done));
+  // Run the main service to make the Put in sync mode
```

Comment:
What does this comment mean?

Reply:
`  RAY_CHECK_OK(gcs_table_storage_->InternalConfigTable().Put(ray::UniqueID::Nil(),                                                             stored_config, on_done));`
is an async function, so we are not sure when the function finished.

`on_done` will be called in main_service. so we need main_service_.run() to block here and once it finished, it'll resume.

previously it's done in [main function ](https://github.com/ray-project/ray/blob/master/src/ray/gcs/gcs_server/gcs_server_main.cc#L58-L84) with promise and a separate thread which I feel is super wired.

Reply:
Maybe rephrase this comment as `Run async operations from InternalConfigTable().Put() above inline.`?